app/code/community/Oyst/OneClick/Model/Payment/Method/Oneclick.php

```diff
@@ -42,4 +44,40 @@ public function getName()
     {
         return self::PAYMENT_METHOD_NAME;
     }
+
+    /**
+     * Refund specified amount for payment.
+     *
+     * @param Varien_Object $payment
+     * @param float $amount
+     *
+     * @return $this
+     */
+    public function refund(Varien_Object $payment, $amount)
+    {
+        /** @var Oyst_OneClick_Helper_Data $helper */
+        $helper = Mage::helper('oyst_oneclick');
+
+        if ($helper->getOpenRefundTransaction($payment)->getId()) {
+            Mage::throwException($helper->__("There is already one credit memo in the queue."));
+        }
+
+        /** @var Oyst_OneClick_Model_Order_ApiWrapper $api */
+        $api = Mage::getModel('oyst_oneclick/order_apiWrapper');
+
+        $response = $api->refund($payment->getLastTransId(), $amount);
+
+        if (!$response) {
```

Comment:
@rus-daniel could you remove the if and use try catch:
```
try {
    /** @var Oyst_OneClick_Model_Order_ApiWrapper $api */
    $api = Mage::getModel('oyst_oneclick/order_apiWrapper');

    $response = $api->refund($payment->getLastTransId(), $amount);
} catch (Exception $e) {
    $orderId = $payment->getOrder()->getId();
    $helper->log(
        sprintf('Unable to refund order %s because the capture transaction does not exist.', $orderId)
    );

    Mage::throwException($helper->__(
        'Impossible to issue a refund transaction because the capture transaction does not exist.'
    ));
}
```